File: Backend/src/models/Transaction.ts
import { model, Schema, Document } from "mongoose";
import { OrderDoc } from "./Order.js";
import { UserDoc } from "./User.js";

export type TransactionDoc = {
  orderId: Schema.Types.ObjectId | OrderDoc;
  userId: Schema.Types.ObjectId | UserDoc;
};

const TransactionSchema = new Schema({
  orderId: {
    type: Schema.Types.ObjectId,
    ref: "Order",
  },
  userId: {
    type: Schema.Types.ObjectId,
    ref: "User",
  },
  paymentMethod: String,
  status: String,
  paymentLog: String,
});

export const Transaction = model("Transaction", TransactionSchema);

File: python/Backend/application/models/Transaction.py
from application.models.db import * 
from application.models.Order import Order
from application.models.User import User
class Transaction(Document):
    orderId= ReferenceField(Order)
    userId= ReferenceField(User, reverse_delete_rule=CASCADE)
    paymentMethod=StringField()
    status=StringField()
    paymentLog=StringField()

    def to_json(self):
        return {
            "orderId": self.orderId,
            "userId": self.userId,
            "paymentMethod": self.paymentMethod,
            "paymentLog": self.paymentLog,
            "status": self.status
        }
    